doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+
+A token has an associated value, which may be empty. If not empty, it is the result of applying a token-specific canonicalization process.
+With the exception of string literals, canonicalization first applies Unicode compatibility normalization NFKC to the token text, then further processing as detailed below.
+
+
+### Comments
+<a id="comments"/>
+
+A comment is introduced by `#` and then extends to the end of the line. It is the only context in which a line end is not regarded simply as whitespace.
+
+> ***comment*** ::= U+0023 `#` NUMBER SIGN _comment-character_* _new-line_
+>
+> _comment_character_ ::= [^\\u{000A}\\u{000B}\\u{000C}\\u{000D}\\u{0085}\\u{2028}\\u{2029}]
+>
+> _new-line_ ::= _cr_ | _lf_ | _cr_ _lf_ | _other-newline_
+>
+> _lf_ ::= U+000A LINE FEED
+>
+> _cr_ ::= U+000D CARRIAGE RETURN
+>
+> _other-new-line_ ::= U+0085 NEXT LINE | U+2028 LINE SEPARATOR | U+2029 PARAGRAPH SEPARATOR
+
+#### Token value
+
+A ***comment*** token has no value.
+
+#### Notes
+
+A _comment-character_ is any character that is not the first character of _new-line_. New line characters and sequences correspond to the [line boundaries requirement RL1.6 of Unicode Regular Expressions](https://www.unicode.org/reports/tr18/#RL1.6).
+
+
+### Whitespace
+<a id="whitespace"/>
+
+In some expressions, where it denotes multiplication, whitespace is mandatory, but otherwise whitespace is not significant. It comprises any non-empty sequence of characters with the White_Space property.
+
+> ***whitespace*** ::= {White_Space}\+
+
+#### Token value
+
+A ***whitespace*** token has no value.
+
+
+### Symbols
+<a id="symbols"/>
+
+Symbols correspond to Arblang identifiers and keywords. Arblang keywords are generally contextual: the interpretation of a symbol as an identifier or a keyword is determined in parsing.
+
+> ***symbol*** ::= _symbol-start_ _symbol-continue_\* | _symbol-degree-prefixed_
+>
+> _symbol-start_ ::= {L} | {Nl} | {Other_ID_Start}
+>
+> _symbol-continue_ ::= _symbol-start_ | {Nd} | {Mn} | {Mc} | {Pc} | _prime-mark_
+>
+> _prime-mark_ ::= U+0027 `'` APOSTROPHE | U+02B9 `ʹ` MODIFIER LETTER PRIME | U+2032 `′` PRIME
+>
+> _symbol-degree-prefixed_ ::= U+00B0 `°` DEGREE SIGN {L}+ | U+2109 `℉` DEGREE FAHRENHEIT | U+2109 `℃` DEGREE CENTIGRADE
+
+#### Token value
+
+Canonicalization substitutes any character in _prime-mark_ with U+0027 APOSTROPHE after performing NFKC normalization.
+
+#### Notes
+
+Identifiers broadly follow the conventions of Python, except:
+* `_` is not a valid _symbol-start_.
+* A ***symbol*** may contain _prime-mark_ characters.
+* Special degree-sign prefixed symbols are permitted, in order to represent temperature scales. Compatibility characters U+2109 `℉` and U+2109 `℃` will be converted to `°F` and `°C` respectively in NFKC normalization.
```

Comment:
We agreed on skipping Temperature Units except `K`, so I think we can skip these symbols as well.

Reply:
If we keep them in the lexical grammar, we can add the units and semantics later without having to change anything here. The degree sign has no other place in the language, so there is little cost in permitting it here.